apis/mysql/v1alpha1/provider_types.go

```diff
@@ -31,9 +31,27 @@ type ProviderConfigSpec struct {
 	// or use preferred to use TLS only when advertised by the server. This is similar
 	// to skip-verify, but additionally allows a fallback to a connection which is
 	// not encrypted. Neither skip-verify nor preferred add any reliable security.
-	// +kubebuilder:validation:Enum="true";skip-verify;preferred
+	// Alternatively, set tls=custom and provide a custom TLS configuration via the tlsConfig field.
+	// +kubebuilder:validation:Enum="true";skip-verify;preferred;custom
 	// +optional
 	TLS *string `json:"tls"`
+
```

Comment:
you _may_ consider enforcing this
https://kubernetes.io/docs/reference/using-api/cel/

Reply:
That would be an interesting exercise, but I have never used CEL, would need to read up on it and learn it, so I'd maybe hope to get away with it! 😅